internal/backend/rest/config.go

```diff
@@ -46,3 +43,37 @@ func ParseConfig(s string) (interface{}, error) {
 	cfg.URL = u
 	return cfg, nil
 }
+
+// StripPassword removes the password from the URL
+// If the repository location cannot be parsed as a valid URL, it will be returned as is
+// (it's because this function is used for logging errors)
+func StripPassword(s string) string {
+	scheme := s[:5]
+	s = prepareURL(s)
+
+	u, err := url.Parse(s)
+	if err != nil {
+		return scheme + s
+	}
+
+	if _, set := u.User.Password(); !set {
+		return scheme + s
+	}
+
+	// A password was set: we replace the UserInfo and rebuild the string from there
+	u.User = url.UserPassword(u.User.Username(), "***")
+	stripped, err := url.QueryUnescape(u.String())
```

Comment:
I'm not sure whether it's a good idea to call `QueryUnescape`?

The go standard library just replaces strings: https://github.com/golang/go/blob/d91d0762c7757c12c7d5f9e2ae3f170d5bd7ba84/src/net/http/client.go#L1004
Or less hacky: Mirror the url.Redacted() method added in go 1.15 (https://golang.org/src/net/url/url.go?s=25219:25250#L852) and replace the password with "xxxxx"

Reply:
I remember why I was doing it, it's because the * was converted into an html character. I've changed to an 'x' instead (like in go 1.15 library)